v-next/hardhat/src/internal/builtin-plugins/console/index.ts

```diff
@@ -0,0 +1,19 @@
+import type { HardhatPlugin } from "@ignored/hardhat-vnext-core/types/plugins";
+
+import { task } from "@ignored/hardhat-vnext-core/config";
+
+const hardhatPlugin: HardhatPlugin = {
+  id: "console",
+  tasks: [
+    task("console", "Opens a hardhat console")
+      .setAction(import.meta.resolve("./task-action.js"))
```

Comment:
In v2 we compile the project before running by default, and have a `--no-compile` flag. We should keep that functionality.

Reply:
correct